src/app/services/contracts.service.ts

```diff
@@ -0,0 +1,156 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {NGXLogger} from 'ngx-logger';
+import {Observable} from 'rxjs';
+import {environment} from '../../environments/environment';
+import {catchError, filter, map} from 'rxjs/operators';
+import {SnackService} from './snack.service';
+import {StaticValues} from '../static/static-values';
+import {Addresses} from '../static/addresses';
+
+interface Result<T> {
+    data: T | Array<T>;
+}
+
+// eslint-disable-next-line
+export interface IsContract{}
```

Comment:
create a file for this interface
better to name ContractI or IContract
add in the interfce this methods
    isStableCoin() {
        return this.underlying?.type === 4 && !this.contract?.name?.match(/CRV/);
    }
    isCRV() {
        return this.underlying?.type === 4 && this.contract?.name?.match(/CRV/);
    }
    isUniLP() {
        return this.underlying?.type === 2 && this.underlying?.name?.match(/^UNI_LP/);
    }
    isOtherLP() {
        return this.underlying?.type === 2 && !this.underlying?.name?.match(/^UNI_LP/);
    }

Reply:
Implementation cannot go in an interface (typescript does not support traits).
Additionally; pool, token, and pair do not have an underlying property, thereby they cannot support this interface.